core/app/models/spree/promotion_rule_user.rb

```diff
@@ -0,0 +1,8 @@
+module Spree
+  class PromotionRuleUser < Spree::Base
+    self.table_name = 'spree_promotion_rules_users'
+
+    belongs_to :promotion_rule, class_name: 'Spree::PromotionRule'
+    belongs_to :user, class_name: Spree::UserClassHandle.new
```

Comment:
An more lazily evaluated form of just calling `Spree.user_class.to_s`, makes initialization more bullet-proof.